src/OAuthClient.php

```diff
@@ -2,403 +2,483 @@
 
 namespace Weble\ZohoClient;
 
-use DateTime;
-use GuzzleHttp\Client;
-use GuzzleHttp\Exception\ClientException;
+use Asad\OAuth2\Client\Provider\Zoho;
+use League\OAuth2\Client\Grant\RefreshToken;
+use League\OAuth2\Client\Provider\AbstractProvider;
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\ResourceOwnerInterface;
+use League\OAuth2\Client\Token\AccessToken;
+use League\OAuth2\Client\Token\AccessTokenInterface;
 use Psr\Cache;
 use Psr\Cache\InvalidArgumentException;
 use Psr\Http\Message\UriInterface;
 use Weble\ZohoClient\Enums\Region;
 use Weble\ZohoClient\Exception\AccessDeniedException;
-use Weble\ZohoClient\Exception\ApiError;
-use Weble\ZohoClient\Exception\CannotGenerateAccessToken;
-use Weble\ZohoClient\Exception\CannotGenerateRefreshToken;
-use Weble\ZohoClient\Exception\GrantCodeNotSetException;
-use Weble\ZohoClient\Exception\InvalidGrantCodeException;
+use Weble\ZohoClient\Exception\AccessTokenNotSet;
+use Weble\ZohoClient\Exception\CannotRevokeRefreshToken;
 use Weble\ZohoClient\Exception\RefreshTokenNotSet;
 
 class OAuthClient
 {
-    const ZOHO_CLIENT_API_URL = 'https://accounts.zoho';
-    const ZOHO_CLIENT_API_PATH = '/oauth/v2';
-
-    /** @var Region */
+    /** @var string */
     protected $region;
-    /** @var Client */
-    protected $client;
     /** @var string|null */
     protected $grantCode;
     /** @var string|null */
     protected $redirectUri;
     /** @var array<string> */
     protected $scopes = ['AaaServer.profile.READ'];
     /** @var bool */
-    protected $offlineMode = false;
+    protected $offlineMode = true;
```

Comment:
Did you intend for offlinemode to be the default of true?

Reply:
Yes, since it's a major version and most of the people use it server to server, this should be the default